src/Core/Rexl.Bind/Operations/Funcs/Text.cs

```diff
@@ -758,3 +758,66 @@ public static string Exec(string src, string remove, string insert)
         return src.Replace(remove, insert);
     }
 }
+
+public sealed partial class TextPadFunc : RexlOper
+{
+    public static readonly TextPadFunc Left = new TextPadFunc(isLeft: true);
+    public static readonly TextPadFunc Right = new TextPadFunc(isLeft: false);
+
+    public readonly bool IsLeft;
+
+    private TextPadFunc(bool isLeft)
+        : base(isFunc: true, new DName(isLeft ? "PadLeft" : "PadRight"), BindUtil.TextNs, 2, 2)
+    {
+        IsLeft = isLeft;
+    }
+
+    protected override ArgTraits GetArgTraitsCore(int carg)
+    {
+        Validation.Assert(SupportsArity(carg));
+        var maskAll = BitSet.GetMask(carg);
+        var maskOpt = maskAll.ClearBit(0);
+        return ArgTraitsLifting.Create(this, carg, maskLiftSeq: maskAll, maskLiftTen: maskAll, maskLiftOpt: maskOpt);
+    }
+
+    protected override (DType, Immutable.Array<DType>) SpecializeTypesCore(InvocationInfo info)
+    {
+        Validation.AssertValue(info);
+        Validation.Assert(SupportsArity(info.Arity));
+        Validation.Assert(info.Arity == 2);
+
+        return (DType.Text, Immutable.Array.Create(DType.Text, DType.I8Req));
+    }
+
+    protected override bool CertifyCore(BndCallNode call, ref bool full)
+    {
+        if (call.Type != DType.Text)
+            return false;
+        var args = call.Args;
+        if (args[0].Type != DType.Text)
+            return false;
+        if (args[1].Type != DType.I8Req)
+            return false;
+        return true;
+    }
+
```

Comment:
It's probably best to include `ReduceCore` in this PR. Otherwise, there is no reason for the `ExecXxx` methods to be in this class - they would normally be in the code generator class.